### ISSUE TITLE
Is it normal for the loss of LLM pre-training not to decrease much after the first epoch?

### ISSUE THREAD

**Issue by wz043** [wz043/llm_train#1]
![Image](https://github.com/user-attachments/assets/6185476f-3e4b-4223-8d9b-65a6d001af4d)
The model parameters are approximately 0.7B, the training length is 512, the batch size is 9, the maximum learning rate is 1e-4, with cosine decay. The text uses the general monkey dataset. Could someone please take a look and see if there are any issues with the code? Thank you.

[inference_model.py](https://github.com/user-attachments/files/22309068/inference_model.py)
[train.py](https://github.com/user-attachments/files/22309067/train.py)
[train_model.py](https://github.com/user-attachments/files/22309066/train_model.py)
